setup.py

```diff
@@ -1,118 +1,144 @@
-#!/usr/bin/env python
-"""Ibis setup module."""
-import pathlib
-import sys
+# -*- coding: utf-8 -*-
+from setuptools import setup
 
-from setuptools import find_packages, setup
-
-import versioneer
-
-LONG_DESCRIPTION = """
-Ibis is a productivity-centric Python big data framework.
-
-See http://ibis-project.org
-"""
-
-VERSION = sys.version_info.major, sys.version_info.minor
-
-impala_requires = ['hdfs>=2.0.16', 'sqlalchemy>=1.3', 'requests']
-impala_requires.append('impyla[kerberos]>=0.15.0')
+packages = [
+    "ibis",
+    "ibis.backends",
+    "ibis.backends.base",
+    "ibis.backends.base.file",
+    "ibis.backends.base.sql",
+    "ibis.backends.base.sql.alchemy",
+    "ibis.backends.base.sql.compiler",
+    "ibis.backends.base.sql.registry",
+    "ibis.backends.clickhouse",
+    "ibis.backends.clickhouse.tests",
+    "ibis.backends.csv",
+    "ibis.backends.csv.tests",
+    "ibis.backends.dask",
+    "ibis.backends.dask.execution",
+    "ibis.backends.dask.tests",
+    "ibis.backends.dask.tests.execution",
+    "ibis.backends.hdf5",
+    "ibis.backends.hdf5.tests",
+    "ibis.backends.impala",
+    "ibis.backends.impala.tests",
+    "ibis.backends.mysql",
+    "ibis.backends.mysql.tests",
+    "ibis.backends.pandas",
+    "ibis.backends.pandas.execution",
+    "ibis.backends.pandas.tests",
+    "ibis.backends.pandas.tests.execution",
+    "ibis.backends.parquet",
+    "ibis.backends.parquet.tests",
+    "ibis.backends.postgres",
+    "ibis.backends.postgres.tests",
+    "ibis.backends.pyspark",
+    "ibis.backends.pyspark.tests",
+    "ibis.backends.sqlite",
+    "ibis.backends.sqlite.tests",
+    "ibis.backends.tests",
+    "ibis.common",
+    "ibis.expr",
+    "ibis.tests",
+    "ibis.tests.expr",
+    "ibis.tests.sql",
+    "ibis.udf",
+]
 
-sqlite_requires = ['sqlalchemy>=1.3']
-postgres_requires = sqlite_requires + ['psycopg2']
-mysql_requires = sqlite_requires + ['pymysql']
+package_data = {"": ["*"]}
 
-kerberos_requires = ['requests-kerberos']
-visualization_requires = ['graphviz']
-clickhouse_requires = [
-    'sqlalchemy<1.4',
-    'clickhouse-sqlalchemy',
-    'clickhouse-driver>=0.1.3',
-    'clickhouse-cityhash',
+install_requires = [
+    "cached_property>=1,<2",
+    "cytoolz>=0.11,<0.12",
+    "multipledispatch>=0.6,<0.7",
+    "numpy>=1,<2",
+    "pandas>=1,<2",
+    "pytz==2021.1",
+    "regex==2021.7.6",
+    "toolz>=0.11,<0.12",
 ]
-hdf5_requires = ['tables>=3.0.0']
-
-parquet_requires = ['pyarrow>=0.12.0']
-spark_requires = ['pyspark>=2.4.3']
 
-geospatial_requires = ['geoalchemy2', 'geopandas', 'shapely']
+extras_require = {
+    "all": [
+        "clickhouse-driver>=0.1,<0.2",
+        "clickhouse-sqlalchemy>=0.1,<0.2",
+        "dask[dataframe,array]>2021.6.0",
+        "geoalchemy2>=0.9,<0.10",
+        "geopandas>=0.9,<0.10",
+        "graphviz>=0.16,<0.17",
+        "hdfs>=2,<3",
+        "impyla[kerberos]==0.18a1",
+        "psycopg2>=2,<3",
+        "pyarrow>=5,<6",
```

Comment:
this is pretty strict here, why not a much earlier version of pyarrow?

Reply:
Mostly because pyarrow tends to break stuff liberally at every release (which is fine, it's major version bump). I can certainly try decreasing the lower bound.

Reply:
What's a good lower bound? 3, 4?

Reply:
i would say >=1 (yes these are not that different), but the release cadence is quite aggressive in pyarrow and this would *force* updating. for newer versions of ibis, can certainly bump.

Reply:
We should try with minimum version `1.0` and bump as needed.

Reply:
This is set to 1.0 now.